cmake/extensions.cmake

```diff
@@ -924,7 +927,16 @@ function(zephyr_linker_sources location)
     message(fatal_error "Must choose valid location for linker snippet.")
   endif()
 
-  foreach(file IN ITEMS ${ARGN})
+  cmake_parse_arguments(L "" "KEY" "" ${ARGN})
+  set(KEY default)
+  if(DEFINED L_KEY)
+    set(KEY ${L_KEY})
+    if(L_KEY MATCHES ".*~.*")
+      message(FATAL_ERROR "Keys are not allowed to contain '~' (${L_KEY})")
```

Comment:
Can I ask why only `~`? Are there no more special cases like `*`?

Reply:
Thanks, this is wrong. Earlier I had a design that used `~` as a list separator. I'll remove this.

Reply:
Done